File: server/models/LoanApplication.js
const mongoose = require("mongoose");

const loanApplicationSchema = new mongoose.Schema(
  {
    msmeId: { type: mongoose.Schema.Types.ObjectId, ref: "MSME", required: true },
    applicant: { type: mongoose.Schema.Types.ObjectId, ref: "User", required: true },
    assignedOfficer: { type: mongoose.Schema.Types.ObjectId, ref: "User" },
    scoreId: { type: mongoose.Schema.Types.ObjectId, ref: "CreditScore", required: true },
    requestedAmount: { type: Number, required: true, min: 0 },
    loanPurpose: {
      type: String,
      enum: ["working_capital", "equipment", "expansion", "other"],
      required: true,
    },
    repaymentTenure: { type: Number }, // months
    status: {
      type: String,
      enum: ["submitted", "under_review", "approved", "rejected", "disbursed"],
      default: "submitted",
    },
    officerRemarks: { type: String },
    approvedAmount: { type: Number },
    decisionDate: { type: Date },
  },
  { timestamps: true }
);

loanApplicationSchema.index({ msmeId: 1 });
loanApplicationSchema.index({ status: 1 });
loanApplicationSchema.index({ assignedOfficer: 1 });

module.exports = mongoose.model("LoanApplication", loanApplicationSchema);

File: server/app/models/loan_application.py
"""LoanApplication document — mirrors the original Mongoose LoanApplication schema."""

from beanie import Document
from pydantic import Field
from typing import Optional, Literal
from datetime import datetime


class LoanApplication(Document):
    msme_id: str
    applicant: str  # User ID
    assigned_officer: Optional[str] = None  # User ID
    score_id: str  # CreditScore ID
    requested_amount: float = Field(ge=0)
    loan_purpose: Literal["working_capital", "equipment", "expansion", "other"]
    repayment_tenure: Optional[int] = None  # months
    status: Literal["submitted", "under_review", "approved", "rejected", "disbursed"] = "submitted"
    officer_remarks: Optional[str] = None
    approved_amount: Optional[float] = None
    decision_date: Optional[datetime] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "loanapplications"

    def to_response_dict(self) -> dict:
        return {
            "_id": str(self.id),
            "msmeId": self.msme_id,
            "applicant": self.applicant,
            "assignedOfficer": self.assigned_officer,
            "scoreId": self.score_id,
            "requestedAmount": self.requested_amount,
            "loanPurpose": self.loan_purpose,
            "repaymentTenure": self.repayment_tenure,
            "status": self.status,
            "officerRemarks": self.officer_remarks,
            "approvedAmount": self.approved_amount,
            "decisionDate": self.decision_date.isoformat() if self.decision_date else None,
            "createdAt": self.created_at.isoformat(),
            "updatedAt": self.updated_at.isoformat(),
        }
